wordpress-mobile/WordPress-Android
# Issue/9666 post list show real progress

Fixes #9666

~**Note: Update branch to `feature/master-post-filters` when #9707 is merged.**~

- since this branch will create a bunch of conflicts with #9657 + it'll need to be updated to take into account the Compact list item view I think it should be reviewed and merged after #9657 is merged -> I'm creating just a draft so we don't forget about it.

Shows real progress as the media upload progresses.
![uploading-media-progress](https://user-images.githubusercontent.com/2261188/56668622-1ee34980-66b0-11e9-85d5-e5cc1d7dc02b.gif)

I've suppressed the item animation in https://github.com/wordpress-mobile/WordPress-Android/commit/6b521a88d05b2d2f1169369a859eff8c0b1df2b8 to prevent the list item from blinking as the upload progresses. However, the featured image is still blinking since it apparently blinks when the `ImageManager.load` method is re-invoked. I can't think of a good and easy solution right now. Any ideas? cc @mzorz @oguzkocer 


To test:
1. Open Post List
2. Create a post
3. Add title and add some images from the local storage
4. Publish the post before the images are uploaded (you might want to set Edge in emulator's networks settings)
5. Notice the progress bar on the item in the list is updated as the upload progresses

Update release notes:

- I haven't updated the release notes as this issue will be released along with the Post Filters project
